packages/apib-parser/lib/adapter.js

```diff
@@ -42,9 +42,46 @@ function parse({
     requireBlueprintName,
   };
 
-  return drafter.parse(source, options);
+  const formatLink = {
+    links: {
+      element: 'array',
+      content: [
+        {
+          element: 'link',
+          meta: {
+            title: {
+              element: 'string',
+              content: 'Apiary Blueprint',
+            },
+          },
+          attributes: {
+            relation: {
+              element: 'string',
+              content: 'via',
+            },
+            href: {
+              element: 'string',
+              content: 'https://apiary.io/blueprint',
+            },
+          },
+        },
+      ],
+    },
+  };
+
+  return drafter.parse(source, options).then((result) => {
+    const refractElement = {};
+    refractElement.element = result.element;
+    refractElement.meta = formatLink;
+    refractElement.content = result.content;
+    return refractElement;
```

Comment:
One caveat here is that if the underlying parser added any other meta or attributes they would go missing. It might be a bit simpler to construct a minim elements from the given result. In `options` there is `namespace` which can be used as a minim namespace. You can get hold of a ParseResult element instance in the same way that the other parsers work. Returning a ParseResult here does not change the behaviour, the API Elements Core logic detects if the returned value is a Minim element, in the cases it is not, it will convert it to be one (thus doing what you'd be doing here).

```js
const { namespace } = options;
const parseResult = namespace.fromRefract(source);
.. new namespace.elements.Link(...);
return parseResult;
```

What do you think?

Reply:
@kylef in `options` there is not `namespace`,  I don't know how to access it here. Seems to me that API Elements are managed internally by the `drafter` function

Reply:
The options passed into the function:

```

function parse({
  source, generateSourceMap, generateMessageBody, generateMessageBodySchema,
  requireBlueprintName,
}) {
```

can be adjusted to add `namespace`.

Reply:
changed